src/descriptor/satisfied_constraints.rs

```diff
@@ -1074,8 +1112,8 @@ mod tests {
         fn from_stack<'stack, 'elem, F>(
             verify_fn: F,
             stack: Stack<'stack>,
-            ms: &'elem Miniscript<bitcoin::PublicKey>,
-        ) -> SatisfiedConstraints<'elem, 'stack, F>
+            ms: &'elem Miniscript<bitcoin::PublicKey, Legacy>,
+        ) -> SatisfiedConstraints<'elem, 'stack, Legacy, F>
```

Comment:
Curious why we use `Legacy` here rather than `Any`

Reply:
As you mentioned in a later comment, we should never actually use `Any` other than whenever required. 
As such, there is no way to create an `Any` type Miniscript from string or decoding, we must specify a `Segwitv0` or `Legacy` context along with it. If we try to create an `Any` type Miniscript, the context checks would map to `unreachable!()`.